airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -34,10 +36,13 @@
     from sqlalchemy.orm import Session
     from sqlalchemy.sql.expression import ColumnOperators
 
+    from airflow.models.dagrun import DagRun
+    from airflow.models.operator import Operator
     from airflow.models.taskinstance import TaskInstance
 
 
-class _UpstreamTIStates(NamedTuple):
+@dataclass
+class _UpstreamTIStates:
```

Comment:
Is this just to have the default? Dataclass is pretty significantly slower and not really worthwhile here since the class is unpacked pretty much immediately. Better to stick to a named tuple.